libcontainer/cgroups/fs2/create.go

```diff
@@ -95,10 +102,18 @@ func CreateCgroupPath(path string, c *configs.Cgroup) (Err error) {
 					}
 				}()
 			}
-			// Write cgroup.type explicitly.
-			// Otherwise ENOTSUP may happen.
-			cgType := filepath.Join(current, "cgroup.type")
-			_ = ioutil.WriteFile(cgType, []byte("threaded"), 0644)
+			// There are 4 types: 'domain', 'threaded', 'domain threaded', and 'domain invalid'.
+			// If got 'domain invalid', we should check whether the current config contains domain controller or not.
+			// If it does not contain domain controller, we can write "threaded" without retuning error.
+			cgTypeFile := filepath.Join(current, "cgroup.type")
+			cgType, _ := ioutil.ReadFile(cgTypeFile)
+			if strings.TrimSpace(string(cgType)) == "domain invalid" {
+				if containsDomainController(c) {
+					return fmt.Errorf("Can't enter cgroup path %s, the cgroup type is 'domain invalid.", current)
```

Comment:
Nit: you have opening single quote before "domain invalid", but do not have a closing one.